src/core/effects/Noise.vue

```diff
@@ -0,0 +1,32 @@
+<script lang="ts" setup>
+import { BlendFunction, NoiseEffect } from 'postprocessing'
+import { useEffect } from '../composables/effect'
+import { makePropWatchersUsingAllProps } from '../../util/prop'
+
+export interface NoiseProps {
+  /**
+   * Whether the noise should be multiplied with the input color.
+   */
+  premultiply?: boolean
+  blendFunction?: BlendFunction
+}
+
+const props = withDefaults(defineProps<NoiseProps>(), {
+  premultiply: false,
+  blendFunction: BlendFunction.SCREEN,
+})
+
+const { pass, effect } = useEffect(() => new NoiseEffect({
+  premultiply: props.premultiply,
+  blendFunction: props.blendFunction,
+}))
+defineExpose({ pass, effect }) // to allow users to modify pass and effect via template ref
+
+makePropWatchersUsingAllProps(
+  props,
```

Comment:
blendFunction should be omitted

Reply:
@Tinoooo  Why should it be omitted? As far as I understand not all the passes accept this prop in case it is being settled on a higher level

Reply:
There is no setter in the pass. Changing the value of `blendFunction` will have no effect. So the watcher would be useless.